src/application/tactics/spin_on_init.c

```diff
@@ -0,0 +1,93 @@
+/*!
+ * @file
+ * @brief
+ */
+
+#include "spin_on_init.h"
+#include "tactic.h"
+#include "tiny_utils.h"
+#include "tuning.h"
+#include "i_tiny_adc.h"
+#include <stdio.h>
+
+enum {
+  max_counts = UINT16_MAX
+};
+
+static void stop_running_timer_expired(tiny_timer_group_t* group, void* context) {
+  reinterpret(self, context, spin_on_init_t*);
+  (void)group;
+  uint8_t stopped;
+  tiny_key_value_store_read(
+    self->key_value_store,
+    self->keys->tactic_stopped,
+    &stopped);
+
+  stopped++;
+
+  tiny_key_value_store_write(
+    self->key_value_store,
+    self->keys->tactic_stopped,
+    &stopped);
+}
+
+static tiny_adc_counts_t potentiometer_counts(spin_on_init_t* self) {
+  tiny_adc_counts_t counts;
+  tiny_key_value_store_read(
+    self->key_value_store,
+    self->keys->knob_counts,
+    &counts);
+  return counts;
+}
+
+static void spin(spin_on_init_t* self, bool spin_right) {
+  motor_power_t left_power = spin_right ? spin_on_init_near_wheel_power : spin_on_init_far_wheel_power;
+  motor_power_t right_power = spin_right ? spin_on_init_far_wheel_power : spin_on_init_near_wheel_power;
+  tiny_key_value_store_write(
+    self->key_value_store,
+    self->keys->left_motor,
+    &left_power);
+  tiny_key_value_store_write(
+    self->key_value_store,
+    self->keys->right_motor,
+    &right_power);
+}
+
+static tiny_timer_ticks_t counts_to_ticks(tiny_adc_counts_t counts) {
+  return (((uint64_t)counts) * (uint64_t)turn_time_360_degree) / (uint64_t)max_counts;
```

Comment:
Shouldn't this range from non-zero to zero back to non-zero? This looks like a simple linear mapping which shouldn't work correctly since right is from 0xFFFF / 2 to 0xFFFF and left is 0 to 0xFFFF / 2.

Reply:
It absolutely should, there's a good chance that this is why I couldn't get it working yesterday

Reply:
should be |counts - mid| * turn_time_180 / max_counts, I think

Reply:
fixed it, good catch!